cirq-core/cirq/protocols/kraus_protocol.py

```diff
@@ -94,6 +95,35 @@ def _has_kraus_(self) -> bool:
         """
 
 
+def _strat_kraus_from_apply_channel(val: Any) -> tuple[np.ndarray, ...] | None:
+    """Attempts to compute a value's Kraus operators via its _apply_channel_ method.
+    This is very expensive (O(16^N)), so only do this as a last resort."""
+    method = getattr(val, '_apply_channel_', None)
+    if method is None:
+        return None
+
+    qid_shape = protocols.qid_shape(val)
+
+    eye = qis.eye_tensor(qid_shape * 2, dtype=np.complex128)
+    superop = protocols.apply_channel(
+        val=val,
+        args=protocols.ApplyChannelArgs(
+            target_tensor=eye,
+            out_buffer=np.ones_like(eye) * float('nan'),
```

Comment:
This creates an extra copy of the `ones_like(eye)` arrays which can be expensive for large tensors.  Please replace with a local variable
```
buffer = np.empty_like(eye)
buffer.fill(float('nan'))
```
above and then use `out_buffer=buffer, auxiliary_buffer0=buffer.copy(), auxiliary_buffer1=buffer.copy())`